### PR TITLE
Support download output of pipeline containing `settings` attribute

### PR DESCRIPTION
**Pull request by lllilithyang**
# Description

https://msdata.visualstudio.com/Vienna/_workitems/edit/2172571/
When download output of a pipeline that submitted with previous sdk, we meet an unexpected attribute: `settings`.
This error occurs in `Job._from_rest_object()`
image.
In this pr we modify settings related logic and add ut to ensure that we can get pipeline without meeting unexpected error. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by kdestin — COMMENTED** [review]
Consider updating the CHANGELOG to note this fix

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ def settings(self) -> PipelineJobSettings:
 
     @settings.setter
     def settings(self, value):
-        if value is not None and not isinstance(value, PipelineJobSettings):
+        if value and not isinstance(value, PipelineJobSettings):
             raise TypeError("settings must be PipelineJobSettings or dict but got {}".format(type(value)))
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py:101]: Seems we don't allow dict here?

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py [hunk 2]
```diff
@@ -97,7 +97,7 @@ def settings(self) -> PipelineJobSettings:
 
     @settings.setter
     def settings(self, value):
-        if value is not None and not isinstance(value, PipelineJobSettings):
+        if value and not isinstance(value, PipelineJobSettings):
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py:100]: why this change resolves the issue?

Comment (lllilithyang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py:100]: In this bug, I find that we would pass an empty dict here and empty dict is not None. Therefore I modify here.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py [hunk 3]
```diff
@@ -97,7 +97,7 @@ def settings(self) -> PipelineJobSettings:
 
     @settings.setter
     def settings(self, value):
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py:99]: We would need to add changelog for named output feature.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py (hunk 1 above)
Comment (lllilithyang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py:101]: Yes. As `settings` is not recommended in v2, we should raise error here if we pass a dict with value. In this bug, we pass an empty dict, therefore we can roughly modify this line to fix this bug.

File: sdk/ml/azure-ai-ml/CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,7 @@
 ### Features Added
 - Added support for `tags` on Compute Resources.
 - Added support for promoting data asset from a workspace to a registry
+- Added Support for registering named asset from job output by specifing named and version settings.
```
Comment (wangchao1230) [inline sdk/ml/azure-ai-ml/CHANGELOG.md:8]: typo:
specifing named and version -> specifying name and version

**Review by wangchao1230 — APPROVED** [review]

File: sdk/ml/azure-ai-ml/CHANGELOG.md (hunk 4 above)
Comment (wangchao1230) [reply sdk/ml/azure-ai-ml/CHANGELOG.md:8]: Support -> support
